MSAL/src/native_auth/public/state_machine/error/SignUpPasswordStartError.swift

```diff
@@ -24,42 +24,60 @@
 
 import Foundation
 
-@objc
+@objcMembers
 public class SignUpPasswordStartError: MSALNativeAuthError {
-    /// An error type indicating the type of error that occurred
-    @objc public let type: SignUpPasswordStartErrorType
+    enum ErrorType: CaseIterable {
+        case browserRequired
+        case userAlreadyExists
+        case invalidPassword
+        case invalidUsername
+        case generalError
+    }
+
+    let type: ErrorType
 
-    init(type: SignUpPasswordStartErrorType, message: String? = nil) {
+    init(type: ErrorType, message: String? = nil) {
         self.type = type
         super.init(message: message)
     }
 
+    /// Describes why an error occurred and provides more information about the error.
     public override var errorDescription: String? {
         if let description = super.errorDescription {
             return description
         }
 
         switch type {
         case .browserRequired:
-            return "Browser required"
+            return MSALNativeAuthErrorMessage.browserRequired
         case .userAlreadyExists:
-            return "User already exists"
+            return MSALNativeAuthErrorMessage.userAlreadyExists
         case .invalidPassword:
-            return "Invalid password"
+            return MSALNativeAuthErrorMessage.invalidPassword
         case .invalidUsername:
-            return "Invalid username"
+            return MSALNativeAuthErrorMessage.invalidUsername
         case .generalError:
-            return "General error"
+            return MSALNativeAuthErrorMessage.generalError
         }
     }
 
-}
+    /// Returns `true` if a browser is required to continue the operation.
+    public var isBrowserRequired: Bool {
+        return type == .browserRequired
+    }
 
-@objc
-public enum SignUpPasswordStartErrorType: Int {
-    case browserRequired
-    case userAlreadyExists
-    case invalidPassword
-    case invalidUsername
-    case generalError
+    /// Returns `true` when the user is trying to register an existing email.
```

Comment:
should we use "username"  instead of "email"  here?

Reply:
Absolutely, updated now. Thank you